### PR TITLE
Update no trainer scripts for multiple-choice

### LINKED ISSUE #18437: Update no_trainer scripts to include gather_for_metrics
### Feature request

🤗 Accelerate has a wrapper to help with distributed metric calculation (a tough problem!), and the `no_trainer` scripts should be updated to include it! 

An example can be seen [here](https://github.com/huggingface/accelerate/blob/main/examples/nlp_example.py#L163-L169), below is an example diff of what the integration would look like:

```diff
-             predictions, references = accelerator.gather((predictions, batch["labels"]))
-             # If we are in a multiprocess environment, the last batch has duplicates
-             if accelerator.num_processes > 1:
-                 if step == len(eval_dataloader) - 1:
-                     predictions = predictions[: len(eval_dataloader.dataset) - samples_seen]
-                     references = references[: len(eval_dataloader.dataset) - samples_seen]
-                 else:
-                     samples_seen += references.shape[0]
+             predictions, references = accelerator.gather_for_metrics((predictions, batch["labels"])) 
```

The list of available scripts to update include:

- [x] examples/pytorch/image-classification/run_image_classification_no_trainer.py
- [x] examples/pytorch/language-modeling/run_clm_no_trainer.py
- [x] examples/pytorch/language-modeling/run_mlm_no_trainer.py
- [x] examples/pytorch/multiple-choice/run_swag_no_trainer.py
- [x] examples/pytorch/question-answering/run_qa_beam_search_no_trainer.py
- [x] examples/pytorch/question_answering/run_qa_no_trainer.py
- [x] examples/pytorch/semantic-segmentation/run_semantic_segmentation_no_trainer.py
- [x] examples/pytorch/speech-pretraining/run_wav2vec2_pretraining_no_trainer.py
- [ ] examples/pytorch/summarization/run_summarization_no_trainer.py

### Motivation

This is a great first issue for someone who wants to learn how to use some of the latest bits in Accelerate and get an easy beginner contribution to the library 🤗 

### Your contribution

If you decide to pick up this issue, feel free to ping myself (@muellerzr), @sgugger, or @pacman100 to review 🤗

### PR DESCRIPTION
**Pull request by kiansierra**
# What does this PR do?
Update `run_swag_no_trainer` example to include `accelarator.gather_metrics`

Related to #18437

I ran the script locally with the following arguments
```
"--dataset_name",
"swag",
"--dataset_config_name",
"regular",
"--model_type",
"bert",
"--tokenizer_name",
"bert-base-uncased"
```
## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@muellerzr , @sgugger, @pacman100 
Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks a lot!
